crates/resolver/src/config.rs

```diff
@@ -445,6 +511,34 @@ impl NameServerConfigGroup {
     pub fn merge(&mut self, mut other: Self) {
         self.append(&mut other)
```

Comment:
Need to consider this function in the face of the new ClientConfig, should mention here which one takes presidence, and then make sure that the other group gets those values. My guess is that we want self to overwrite the other's client_configs.

Reply:
This is the first time that I use this library but my impression was that this is a very specific mutator that does not provide functionality that could not be achieved with the constructors ..

Reply:
That's possible. I'm open to removing this, and instead using constructors.